service/role/service.go

```diff
@@ -0,0 +1,114 @@
+package role
+
+import (
+	"fmt"
+
+	"github.com/giantswarm/microerror"
+	vaultclient "github.com/hashicorp/vault/api"
+)
+
+// Config defines configurable aspects (such as dependencies) of this service.
+type Config struct {
+	// Config
+	PKIMountpoint string
+
+	// Dependencies.
+	VaultClient *vaultclient.Client
+}
+
+// DefaultConfig returns a default configuration that can be used to create this service.
+func DefaultConfig() Config {
+	vaultClientConfig := vaultclient.DefaultConfig()
+	vaultClientConfig.Address = "http://127.0.0.1:8200"
+	vaultClient, err := vaultclient.NewClient(vaultClientConfig)
+	if err != nil {
+		panic(err)
+	}
+
+	config := Config{
+		// Dependencies.
+		VaultClient: vaultClient,
```

Comment:
We recently do not define default dependencies anymore since this might cause weird situations in which we think everything is configured properly but in fact it is not. A classic is to forget to configure a dependency. Then the default is used and we only notice when it is too late instead of getting the "invalid config" error.